web/ai_service.py

```
@@ -0,0 +1,141 @@
+            highlighted = highlight(code, lexer, formatter)
+            return f'<div class="code-block">{highlighted}</div>'
+        except:
+            return f'<pre><code class="language-{language}">{code}</code></pre>'
+
+    def _format_math_expression(self, expression: str) -> str:
+        """Format mathematical expressions using MathJax delimiters."""
+        # Handle inline math
+        expression = re.sub(r'\$(.+?)\$', r'\\[\1\\]', expression)
+        # Handle display math
+        expression = re.sub(r'\$\$(.+?)\$\$', r'\\[\1\\]', expression)
+        return expression
```

Ruff (0.8.2)

Unnecessary assignment to `expression` before `return` statement

Remove unnecessary assignment

(RET504)